src/compute/src/render/reduce.rs

```diff
@@ -721,103 +738,75 @@ where
-                .map(|(values, _cnt)| values[aggr_index].iter().next().unwrap());
-            output.push(Row::pack_slice(&[func.eval(iter, &RowArena::new())]));
-        }
-        // We only want to arrange the parts of the input that are not part of the output.
-        // More specifically, we want to arrange it so that `input.concat(&output.negate())`
-        // gives us the intended value of this aggregate function. Also we assume that regardless
-        // of the multiplicity of the final result in the input, we only want to have one copy
-        // in the output.
-        target.push((logic(output), -1));
-        target.extend(
-            source
-                .iter()
-                .map(|(values, cnt)| (logic((*values).clone()), *cnt)),
-        );
-    }
-
-    let (negated_output, stage_errs) = if validating {
-        let (negated_output, errs) = arranged_input
-        .reduce_abelian::<_,RowSpine<_,Result<_,_>,_,_>>("Reduced Fallibly MinsMaxesHierarchical", {
+    let negated_output = arranged_input
+        .reduce_abelian::<_, RowSpine<_, _, _, _>>(
+            "Reduced Fallibly MinsMaxesHierarchical",
             move |key, source, target| {
-                // Should negative accumulations reach us, we should loudly complain.
-                if source.iter().any(|(_val, cnt)| cnt <= &0) {
-                    for (val, cnt) in source.iter() {
-                        // XXX: This reports user data, which we perhaps should not do!
-                        if *cnt <= 0 {
-                            warn!("[customer-data] Non-positive accumulation in MinsMaxesHierarchical: \
-                                key: {key:?}\tvalue: {val:?}\tcount: {cnt:?} in dataflow {debug_name}");
-                            soft_assert_or_log!(
-                                false,
-                                "Non-positive accumulation in MinsMaxesHierarchical"
-                            );
+                if R::validating() {
+                    // Should negative accumulations reach us, we should loudly complain.
+                    for (value, count) in source.iter() {
+                        if count.is_positive() {
+                            continue;
                         }
+                        let message = "Non-positive accumulation in MinsMaxesHierarchical";
+                        warn!(
+                            ?key,
+                            ?value,
+                            ?count,
+                            debug_name,
+                            "[customer-data] {message}"
+                        );
+                        error!("{message}");
+                        // After complaining, output an error here so that we can eventually
+                        // report it in an error stream.
+                        // XXX WIP target.push((R::error((key, value)), -1));
+                        target.push((R::error(key.clone()), -1));
+                        return;
```

Comment:
This changes how the error handling behaves. Previously we reported all invalid counts we found, the new version only reports the first instance. Is that intended?